core/raftlease/fsm.go

```diff
@@ -271,6 +320,7 @@ func (f *FSM) Restore(reader io.ReadCloser) error {
 type Snapshot struct {
 	Version    int                           `yaml:"version"`
 	Entries    map[SnapshotKey]SnapshotEntry `yaml:"entries"`
+	Pinned     map[SnapshotKey]bool          `yaml:"pinned"`
```

Comment:
It'd make more sense to make this a slice of keys - it's really a set, and you don't need to do lookups in the snapshot like you do in the running FSM.

Reply:
Done.